js/ui/shellDBus.js

```diff
@@ -534,25 +545,22 @@ const AppStoreService = new Lang.Class({
     },
 
     AddAppIfNotVisible: function(id) {
-        let visibleIcons = IconGridLayout.layout.getIcons(IconGridLayout.DESKTOP_GRID_ID);
-        let isIconVisible = visibleIcons.indexOf(id) !== -1;
-
-        if (!isIconVisible) {
-            let eventRecorder = EosMetrics.EventRecorder.get_default();
-            let appId = new GLib.Variant('s', id);
-            eventRecorder.record_event(SHELL_APP_ADDED_EVENT, appId);
-        }
+        let iconWasVisible = _iconIsVisibleOnDesktop(id);
 
         if (IconGridLayout.layout.iconIsFolder(id))
             return;
 
         IconGridLayout.layout.appendIcon(id, IconGridLayout.DESKTOP_GRID_ID);
+
+        if (!iconWasVisible) {
```

Comment:
Same comment about the braces here. Also, is there any reason you're moving this after the early return `iconIsFolder()`?

Reply:
FWIW, it looks good to me, I think it does not make any difference in practical terms

Reply:
Ah - yeah I spoke to Cosimo about this yesterday and reporting the metric *before* doing the action is a broken behaviour - we don't want to report a metric if it would have been blocked by the application logic.